vulkan_helpers/helper_functions.cpp

```diff
@@ -1295,15 +1296,30 @@ VkDescriptorSetLayout CreateDescriptorSetLayout(
 }
 
 // Creates a default pipeline cache, it does not load anything from disk.
-VkPipelineCache CreateDefaultPipelineCache(VkDevice* device) {
+VkPipelineCache CreateDefaultPipelineCache(VkDevice* device, const entry::EntryData* entry_data) {
   ::VkPipelineCache cache = VK_NULL_HANDLE;
+  void* initial_data = nullptr;
+  size_t initial_size = 0;
+  std::vector<char> buffer;
+  if (entry_data->load_pipeline_cache()) {
+    std::ifstream in_file(entry_data->load_pipeline_cache(), std::ios::binary | std::ios::ate);
+    initial_size = in_file.tellg();
+    in_file.seekg(0, std::ios::beg);
+    buffer.resize(initial_size);
+    in_file.read(buffer.data(), initial_size);
+    LOG_ASSERT(==, entry_data->logger(), false, in_file.bad());
```

Comment:
Is it accurate to say we expect a failure here if the pipeline cache to load doesn't exist?

Reply:
Yes, If someone specified a pipeline-cache via command-line flag, and we are not able to open/load it, i would say it makes a lot of sense to fail.